### ISSUE TITLE
about func prepare_zigzag_ring_attn_inputs

### ISSUE THREAD

**Issue by peiliu0408** [EvolvingLMMs-Lab/LongVA#14]
as mentioned in title,      

original code:
`def extract_local(value, rank, world_size, device, dim=1):

    value_chunks = value.chunk(2 * world_size, dim=dim)

    local_value = torch.cat( [value_chunks[rank], value_chunks[2 * world_size - rank - 1]], dim=dim)

    return local_value.to(device)
`

assert world_size = 8, therefor, rank = 0, get a slice of input_ids: 

` local_value = torch.cat([value_chunks[0], value_chunks[2 * 8 - 0 - 1]], dim=dim)`

which means(rank = 0), get the leftest and rightest slices of chunked inputs.

why rank=0, get right 2 chunks?
